tests/test_cli_compile.py

```diff
@@ -2190,6 +2190,70 @@ def test_multiple_extras(fake_dists, runner, make_module, fname, content, extra_
     assert "extra ==" not in out.stderr
 
 
+@pytest.mark.network
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_all_extras(fake_dists, runner, make_module, fname, content):
+    """
+    Test passing `--all-extras` includes all applicable extras.
+    """
+    meta_path = make_module(fname=fname, content=content)
+    out = runner.invoke(
+        cli,
+        [
+            "-n",
+            "--all-extras",
+            "--find-links",
+            fake_dists,
+            "--no-annotate",
+            "--no-emit-options",
+            "--no-header",
+            meta_path,
+        ],
+    )
+    assert out.exit_code == 0, out.stderr
+    assert (
+        dedent(
+            """\
+        small-fake-a==0.1
+        small-fake-b==0.2
+        small-fake-c==0.3
+        small-fake-d==0.4
+        small-fake-e==0.5
+        small-fake-f==0.6
+        Dry-run, so nothing updated.
+        """
```

Comment:
This should probably be indented.
```suggestion
            small-fake-a==0.1
            small-fake-b==0.2
            small-fake-c==0.3
            small-fake-d==0.4
            small-fake-e==0.5
            small-fake-f==0.6
            Dry-run, so nothing updated.
            """
```